rfcs/rfc_001.md

```diff
@@ -0,0 +1,161 @@
+- No change in the backend protocol;
+- integrate seemlesly with the existing backend and deployment mechanism;
+- Allow completly independant frontend (GeoGirafe, MasterPortal, Uzuverse, other);
+
+### Simplify Ngeo/Gmf build tools and make the code standard
+
+- Limit breaking changes for existing users;
+- Remove / simplify code;
+- Distribute the Ngeo/GMF library as pure ES6 + CSS package;
+
+### Allow to easily extend Ngeo/Gmf
+
+- Ability to use any build system, frontend framework, or package and in any version;
+- Expose some states on a case-by-case;
+- Document and demo this opt-in mechanism;
+
+
+## Technical proposition part 1: allow alternative frontends
+
+### Creation of a custom frontend
+
+There will be a `frontend` directory at the root of the Geomapfish project. This directory will contain the custom frontend project.
+
+In this folder the developer should add everything needed to build the frontend like a `package.json`, the custom components, the build tools, the interface html files, etc.
+
+This folder will contain a `Dockerfile`, in charge of building this custom frontend project.
+The content is up to the developer, the only constraint is to put the generated files in the `/etc/static-frontend` directory.
+
+Each interface html file must contain the header metadata `dynamicUrl`, pointing to the used `dynamic.json` like: `/dynamic.json` (used by ngeo).
+These interfaces will be served by the backend on `/` and `/theme/<name>`.
+
+The other files in the `frontend` directory will be served by the backend on the path `/static-frontend/<file>`.
+These paths should be correctly referenced in the html files.
+
+Note: above I consider the entry point as `/` but it should be correcty managed by the frontend Docker image if needed.
+
+Note: the i18n should be managed by the frontend project, the backend will only provide language files with localized string related to the backend (like the layer names).
+
+Default custom frontend: the demo will provide a working example, in addition, documentation will be added in the GeoMapFish documentation.
+
+### Integration of the custom frontend
+
+In `c2cgeoportal` we will add a new static route to serve the frontend files from `/etc/static-frontend` directory on the `/static-frontend` path.
+
+In the `c2cgeoportal` configuration we will add a new interface type `custom` (currently we have `ngeo` and `canvas`).
+
+The interface will be defined as:
+
+```yaml
+name: desktop  # The name of the interface
+type: custom
+file: desktop.html  # The html file name
+```
+
+If needed a mako boolean can be added to the interface definition if we need to render the html file with mako.
+
+If needed the file can be a glob pattern if there is an hash in the filename.
+
+
+## Technical proposition part 2: simplify Ngeo/Gmf build tools and make the code standard
+
+Iteratively simplify the existing build tool based on Webpack 4.
+
+### Resolve / stop using ngInject annotation for dependency injection
```

Comment:
The rest of the file is only related to ngeo I think. 
The changes described won't have any impact on a custom frontend, are they?

Reply:
Yes